elephant/statistics.py

```diff
@@ -240,3 +243,283 @@ def lv(v):
+                     'exponential', 'alpha'])
+    if form in forms_verbose:
+        form = forms_abbreviated[forms_verbose == form][0]
+
+    assert form.upper() in ('BOX', 'TRI', 'GAU', 'EPA', 'EXP', 'ALP'), \
+    "form must be one of either 'BOX','TRI','GAU','EPA','EXP' or 'ALP'!"
+
+    assert direction in (1, -1), "direction must be either 1 or -1"
+
+    # conversion to SI units (s)
+    SI_sigma = sigma.rescale('s').magnitude
+    SI_time_stamp_resolution = sampling_period.rescale('s').magnitude
+
+    norm = 1./SI_time_stamp_resolution
+
+    if form.upper() == 'BOX':
+        w = 2.0 * SI_sigma * np.sqrt(3)
+        # always odd number of bins
+        width = 2 * np.floor(w / 2.0 / SI_time_stamp_resolution) + 1
+        height = 1. / width
+        kernel = np.ones((1, width)) * height  # area = 1
+
+    elif form.upper() == 'TRI':
+        w = 2 * SI_sigma * np.sqrt(6)
+        halfwidth = np.floor(w / 2.0 / SI_time_stamp_resolution)
+        trileft = np.arange(1, halfwidth + 2)
+        triright = np.arange(halfwidth, 0, -1)  # odd number of bins
+        triangle = np.append(trileft, triright)
+        kernel = triangle / triangle.sum()  # area = 1
+
+    elif form.upper() == 'EPA':
+        w = 2.0 * SI_sigma * np.sqrt(5)
+        halfwidth = np.floor(w / 2.0 / SI_time_stamp_resolution)
+        base = np.arange(-halfwidth, halfwidth + 1)
+        parabula = base**2
+        epanech = parabula.max() - parabula  # inverse parabula
+        kernel = epanech / epanech.sum()  # area = 1
+
+    elif form.upper() == 'GAU':
+        w = 2.0 * SI_sigma * 2.7  # > 99% of distribution weight
+        halfwidth = np.floor(w / 2.0 / SI_time_stamp_resolution)  # always odd
+        base = np.arange(-halfwidth, halfwidth + 1) * SI_time_stamp_resolution
+        g = np.exp(
+            -(base**2) / 2.0 / SI_sigma**2) / SI_sigma / np.sqrt(2.0 * np.pi)
+        kernel = g / g.sum()
+
+    elif form.upper() == 'ALP':
+        w = 5.0 * SI_sigma
+        alpha = np.arange(
+            1, (
+                2.0 * np.floor(w / SI_time_stamp_resolution / 2.0) + 1) +
+            1) * SI_time_stamp_resolution
+        alpha = (2.0 / SI_sigma**2) * alpha * np.exp(
+            -alpha * np.sqrt(2) / SI_sigma)
+        kernel = alpha / alpha.sum()  # normalization
+        if direction == -1:
+            kernel = np.flipud(kernel)
+
+    elif form.upper() == 'EXP':
+        w = 5.0 * SI_sigma
+        expo = np.arange(
+            1, (
+                2.0 * np.floor(w / SI_time_stamp_resolution / 2.0) + 1) +
+            1) * SI_time_stamp_resolution
+        expo = np.exp(-expo / SI_sigma)
+        kernel = expo / expo.sum()
+        if direction == -1:
+            kernel = np.flipud(kernel)
+
+    kernel = kernel.ravel()
+    m_idx = np.nonzero(kernel.cumsum() >= 0.5)[0].min()
+
+    return kernel, norm, m_idx
+
+
+def instantaneous_rate(spiketrain, sampling_period, form, sigma, m_idx=None,
```

Comment:
This can perhaps wait until later (after merging this PR), but it might be useful to have this function accept a list of spiktrains, and to calculate the rate for all of them at once.